sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpElementAccessTracker.cs

```diff
@@ -55,5 +55,28 @@ public CSharpElementAccessTracker(IAnalyzerConfiguration analyzerConfiguration,
                     constantValue.Value is string constant &&
                     constant == value;
             };
+
+        #region Syntax-level checking methods
+
+        public override ElementAccessCondition MatchSetter() =>
+            (context) => ((ExpressionSyntax)context.Expression).IsLeftSideOfAssignment();
+
+        public override ElementAccessCondition MatchProperty(MemberDescriptor member)
+        {
+            return (context) => {
+                var accessSyntax = context.Expression as ElementAccessExpressionSyntax;
+                if(accessSyntax.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression))
```

Comment:
it may be null after cast, you can replace the if with

```
if (context.Expression is ElementAccessExpressionSyntax accessSyntax &&
    accessSyntax.Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression))
{}
```
this ensures the null-check